### PR TITLE
Chore: Update "airbyte-lib" CLI references with newer PyAirbyte CLI refs (`pyab validate`)

### PR DESCRIPTION
**Pull request by aaronsteers**
## What

The PyAirbyte `validate.py` module was created entirely to help validate that connectors had a valid CLI and that they could be installed pythonically (if "pythonically" is a word 😅 ). The code in that module is entirely dead code, except in that it gets invoked by `airbyte-ci` to make sure (1) we can successfully install the connector and (2) that we can invoke it via CLI and get it to run the `spec` operation successfully.

This PR bumps the version of PyAirbyte used in these tests and replaces the legacy validation test with the newer CLI `validate` command - added this week in PyAirbyte `v0.18.0`.

The new CLI uses the same code paths that PyAirbyt uses normally, which will allow us to remove the legacy `validate.py` file in PyAirbyte - in favor of the more well-traversed and well-tested code paths of the main PyAirbyte featureset.

## How

- Bump the pinned PyAirbyte version.
- Update the CLI syntax to match the new `pyab validate` command.
- Update some other stale refs that still mentioned (codename) `airbyte-lib`.

## User Impact

None

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by aaronsteers** [comment]
@alafanechere - I didn't look deeply into these tests. In theory, this is mostly a no-op, replacing an older legacy CLI with a new one - but let me know what you think in regards to testing strategy.

My hope is that if this works well, we can also wrap the `pyab benchmark` CLI next - to get perf benchmarks included in `airbyte-ci`.

**Comment by aaronsteers** [comment]
Good news is that it failed on first pass due to output text mismatch. Meaning, the codepath appears to be sufficiently tested. I've updated the output text check, and expect this will succeed CI tests on next pass.

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/python_connectors.py [hunk 1]
```diff
@@ -215,7 +215,14 @@ async def _run(self, connector_under_test: Container) -> StepResult:
 
         test_environment = await self.install_testing_environment(with_poetry(self.context))
         test_execution = test_environment.with_(
-            hacks.never_fail_exec(["airbyte-lib-validate-source", "--connector-dir", ".", "--validate-install-only"])
+            hacks.never_fail_exec(
+                [
+                    "pyab",
+                    "validate",
+                    f"--connector={self.context.connector.technical_name}",
+                    "--pip-url='.'",
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/python_connectors.py:223]: What's the meaning of `--pip-url='.'` ?

**Review by alafanechere — APPROVED** [review]
This looks like a clean and legit change! Thank you @aaronsteers .
Can you please bump the `pipelines` package version in `pyproject.toml` and add a changelog entry to the readme? 🙏

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/python_connectors.py (hunk 1 above)
Comment (aaronsteers) [reply airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/python_connectors.py:223]: Pip URL of "." just means to install with `pip install ." - meaning install the Python package from the current working directory. 

This is meant to keep parity with the previous validate command while also supporting other install sources, such as a named PyPi reference or a versioned PyPi reference, or a guy URL, etc

**Comment by aaronsteers** [comment]
I've updated bumped the version number in pyproject.toml and I've added a changelog entry. ✅

**Pull request merged**